### DISCUSSION TITLE
Unknown crashes in Firebase Crashlytics console

### DISCUSSION THREAD

**Discussion by Minishlink** [invertase/react-native-firebase#4327] [Q&A] [answered]
Hi there,

Let me know if this is not the right place for this message. :)
On our app (50k~ daily users) we have many unknown crashes from shared libraries or weird apk : base.apk (4.5k crashes), libjsc.so (304), libwebviewchromium.so, libc.so, Chrome.apk... And I'm having a hard time figuring out the root cause or even reproducing it.

They're not tied to any specific Android version or model, here are the average stats:

* ~50% samsung
* ~40% android 10
* enough memory and storage available
* 0% in background
* 0% jailbroken
* 100% proximity sensor triggered <= what is this?

The stacktrace is not readable at all, is it a misconfiguration of Firebase? We have correct stacktraces for other more common crashes. Maybe it's NDK related?

Here are examples of stacktraces:

```
Crashed: Thread :  SIGSEGV  0x0000000000000000
#00 pc 0xd8291dcc base.apk 
#01 pc 0x14728136  
#02 pc 0x14728136  
#03 pc 0xd757c875 base.apk 
#04 pc 0xc48ea843 base.odex 
#05 pc 0x1336630e  
#06 pc 0x6fea70ee  
#07 pc 0x71ed3153 boot-framework.oat 
#08 pc 0xe805dca2  
#09 pc 0x1325c516  
#10 pc 0x14728136  
#11 pc 0x6fd32c66  
#12 pc 0x6fea70ee  
#13 pc 0x14728136  
#14 pc 0x14728136  
#15 pc 0x133c0626  
#16 pc 0x6fea70ee  
#17 pc 0xc4d04d49 base.odex 
#18 pc 0xe7d936a6  
#19 pc 0x14728136  
#20 pc 0x704c2586  
#21 pc 0x1498a556  
#22 pc 0x13d5a64e  
#23 pc 0x14728136  
#24 pc 0x133c0626  
#25 pc 0x6fea70ee  
#26 pc 0x71f6cb83 boot-framework.oat 
#27 pc 0x70281b12  
#28 pc 0x145f7c26  
#29 pc 0x6fe922ce  
#30 pc 0x1336630e  
#31 pc 0x12fc39be  
#32 pc 0x1498a556  
#33 pc 0x13d5a64e  
#34 pc 0x13d5a64e  
#35 pc 0x72449851 boot-framework.oat 
#36 pc 0x703956fa  
#37 pc 0x6fbeb286  
#38 pc 0x14dc21ce  
#39 pc 0x1498a556  
#40 pc 0x7244bf5b boot-framework.oat 
#41 pc 0x703b2242  
#42 pc 0x14dc21a6  
#43 pc 0x6ff94df6  
#44 pc 0x7104af63  
#45 pc 0x6ffaee46  
#46 pc 0x6f79d5de  
#47 pc 0x14dc21a6  
#48 pc 0x6f9c9dae  
#49 pc 0x6ff9811e  
```

```
Crashed: Thread :  SIGSEGV  0x0000000000000010
#00 pc 0x719b39b748 libjsc.so 
#01 pc 0x719b3e7fe8 libjsc.so 
#02 pc 0x729d1bbba8 libc.so 
```

```
Crashed: Thread :  SIGSEGV  0x0000005500000002
#00 pc 0x7f7131d0b0 libwebviewchromium.so 
#01 pc 0x7f71c69530 libwebviewchromium.so 
#02 pc 0x7f68d45ddc  
#03 pc 0x7f8add9704 libc.so 
#04 pc 0x7f6927e44c  
#05 pc 0x739c0424 system@framework@boot.oat 
#06 pc 0x12cec0fc  
#07 pc 0x7f86b82428 libart.so 
```

```
Crashed: Thread :  SIGSEGV  0x00014ff4a9be7bfd
#00 pc 0x746ba50698  
#01 pc 0x746a989848  
#02 pc 0x746c5b3be4  
#03 pc 0x74cb36a48c libart.so 
#04 pc 0x7416066664  
#05 pc 0x7416066664  
#06 pc 0x741687aa2e  
#07 pc 0x7416243b78  
#08 pc 0x741687aa2e  
#09 pc 0x74cb36b334 libart.so 
#10 pc 0x741687aa2e  
#11 pc 0x74cb37a06c libart.so 
#12 pc 0x6ffa854c  
#13 pc 0x74cb514ca0 libart.so 
#14 pc 0x6ffa854c  
#15 pc 0x74cb50ff80 libart.so 
#16 pc 0x74167ac99a  
#17 pc 0x6fdbbab4  
#18 pc 0x74cb7cf860 libart.so 
#19 pc 0x6ffa8550  
#20 pc 0x6fdbbab4  
#21 pc 0x74cb7d4fc4 libart.so 
#22 pc 0x74cb3620fc libart.so 
#23 pc 0x74cb365a14 libart.so 
#24 pc 0x74cb519a6c libart.so 
#25 pc 0x74cb4e58ec libart.so 
#26 pc 0x74cb7bff94 libart.so 
#27 pc 0x6fdbbab4  
#28 pc 0x74cb7bf6cc libart.so 
#29 pc 0x6fdbbab4  
#30 pc 0x74cb7bf334 libart.so 
#31 pc 0x74cb3b9c2c libart.so 
#32 pc 0x74cb7be164 libart.so 
#33 pc 0x704711fc boot.oat 
#34 pc 0x74cb374468 libart.so 
#35 pc 0x6f662e44  
#36 pc 0x70495ffc boot.oat 
#37 pc 0x7103e114 boot-framework.oat 
#38 pc 0x6fe0fb8c  
#39 pc 0x710416b8 boot-framework.oat 
#40 pc 0x6fe09614  
#41 pc 0x6f93030c  
#42 pc 0x6f4d0b64  
#43 pc 0x6f8245fc  
#44 pc 0x745d9b6ffc base.odex 
#45 pc 0x9d9f359c  
#46 pc 0x74cb156f4c  
#47 pc 0x7055d568 boot.oat 
#48 pc 0x6f5f10fc  
#49 pc 0x74cb36b334 libart.so
```

```
Crashed: Thread :  SIGTRAP  0x0000007dda353950
#00 pc 0x7dda353950 Chrome.apk 
#01 pc 0x7dda35392c Chrome.apk 
#02 pc 0x7dda3537cc Chrome.apk 
#03 pc 0x7ddaad26cc Chrome.apk
```

Current configuration is React Native 0.62.2, no hermes, rnfirebase 8.2.0 / rn-firebase crashlytics 8.1.2

Thanks for your help 🙏

**Comment by mikehardy** [comment, marked as answer]
Things like that are really frustrating aren't they? I have an Android native app with 1.7MM users (AnkiDroid, open source flash cards so it's worth precisely $0 but makes for good knowledge of Android crashes) and you wouldn't believe the stacks we get in Chrome, it manages to fail in endlessly surprising ways. That's what these stacks look like to me mostly - the base.apk and chromium ones at least. libjsc is crashing in the JSC javascript interpreter used by react-native but it's native - it's not a coding error on your part.

There's nothing you can do about the stacks being not-understandable unless you turn on NDK crash support for crashlytics but that's very hard to do because then you have to provide paths to the stripped and unstripped native libraries, and you don't get access to those for Chrome so it doesn't help.

My biggest advice is to update to current stable in case the JSC has improved (or perhaps try swapping in V8 or Hermes - latest Hermes works with RNFB from what we understand) and be careful with your memory usage - use the Android native profiler to make sure you aren't loading huge bitmaps and fonts and things as that's a classic crasher, and especially if you do so in WebViews.

Sorry this is vague but I have had the same experience of receiving these stacks and there isn't much actionable unfortunately unless you can get a reproduction case, and even then it's mostly so you can implement a workaround.

**Reply by Minishlink** [reply]
They are frustrating! 😉

Thanks for your insights, that's cool to know I'm not alone with these crashes :)

Apparently the looks of the stacktraces might be because we enabled Android App Bundle. https://stackoverflow.com/questions/55966582/google-play-crash-logs-not-symbolicated-with-android-app-bundle
But well I'm not sure I want to make the app size bigger just to look at those crashes...

Will update RN and JSC, or test Hermes, thanks!

**Reply by mikehardy** [reply]
They push app bundle pretty hard and yes, it has similar problems to "bitcode-enabled" iOS builds in that it re-munges your so unless there is a way to then download (from Google Play) and upload (to Google Firebase) the new symbol tables I'm not sure there's anything you can do. We still use APKs with ABI splits in the app I mention and get worthless symbols for Chrome crashes anyway, for what it's worth :-) 

I wouldn't skip the Android Studio profiling step and hunting for a real reproduction - for the few crashes that were attributable to our use of WebView for instance, we did find a reproduction in form of loading a very heavy CJK font in combo with images exhausting memory, and we found that implementing the WebView termination handling API was a big improvement: https://developer.android.com/guide/webapps/managing-webview#termination-handle / https://github.com/react-native-webview/react-native-webview/blob/master/docs/Reference.md#onRenderProcessGone

**Reply by mikehardy** [reply]
(I am making the assumption that with lots of Chrome crashes you are using a WebView, and that it's crashing - I'll also mention I understand those are assumptions and if I'm not correct there then go ahead and ignore the suggestions based on them of course, they'll be ineffective)
